network/p2p/blob/blob_service.go

```diff
@@ -238,3 +238,93 @@ func (r *rateLimitedBlockStore) Get(ctx context.Context, c cid.Cid) (blocks.Bloc
 
 	return r.Blockstore.Get(ctx, c)
 }
+
+type Tracer struct {
+	logger zerolog.Logger
+}
+
+func NewTracer(logger zerolog.Logger) *Tracer {
+	return &Tracer{
+		logger,
+	}
+}
+
+func (t *Tracer) MessageReceived(pid peer.ID, msg bsmsg.BitSwapMessage) {
```

Comment:
the logic in these two methods is basically the same. can you refactor so there's less code duplication?